### ISSUE TITLE
-D__FILENAME__=\\\"filename.c\\\"  in compile_commands.json is not parsed properly

### ISSUE THREAD

**Issue by schultetwin1** [microsoft/vscode-cpptools#1795]
We use cmake to generate a \_\_FILENAME\_\_ definition for each file that is passed into the command line. \_\_FILENAME\_\_ is the same as \_\_FILE\_\_ without the full path. In compile_commands.json this will show up in the command as -D\_\_FILENAME\_\_=\\\\\\\"filename.c\\\\\\\".  The first backslash escapes the second backslash and the third backslash escapes the ". This is so that on the command line it will show up as -D__FILENAME__=\\"filename.c\\" so that the quotes are escaped for the command.

In vscode, the intellisense thinks \_\_FILENAME\_\_ is \\"filename.c\\" instead of "filename.c". This causes a red squiggle under the \_\_FILENAME\_\_ macro.

A quick repro of this issue is to add the following to a CMakeLists.txt
```cmake
set(FILENAME filename.c)
target_compile_definitions(target PUBLIC __FILENAME__="${FILENAME}") 
```

**Comment by huww98** [comment]
I have the same problem with my ros project.

`compile_commands.json` contains something like `-DROS_PACKAGE_NAME=\\\"my_package_name\\\"` and VSCode show errors on every rosconsole calls (like `ROS_FATAL("some message.");`). If I change it to  `-DROS_PACKAGE_NAME=\"my_package_name\"`, then everything works.

**Comment by bobbrow** [comment]
Thanks for reporting this.

**Comment by sean-mcmanus** [comment]
I hit this myself last week (from the repro project in https://github.com/Microsoft/vscode-cpptools/issues/1777 ), but I didn't realize it was a bug.

**Comment by bobbrow** [comment]
I'm testing a fix for this right now.

**Comment by schultetwin1** [comment]
Awesome, thank you!

**Comment by bobbrow** [comment]
A preview of this is available in the [insiders build](https://github.com/Microsoft/vscode-cpptools/releases/tag/v0.17.0-insiders) if you would like to try it out and provide feedback.

**Comment by schultetwin1** [comment]
Just tried the preview. It worked great, thank you!

**Comment by bobbrow** [comment]
This is fixed in 0.17.0.
